### PR TITLE
Update for 2022

### PR DESCRIPTION
**Pull request by dlouhak**
- Update the main translations for the survey State of CSS 2022 according to [`locale-en-US` repository](https://github.com/Devographics/locale-en-US/).
- I hope to have time to complete `results.yml` in the future. Currently it's WIP and not included in this PR.

### CODE REVIEW

File: common.yml [hunk 1]
```diff
@@ -348,11 +383,11 @@ translations:
     - key: options.age.range_less_than_10
       t: Méně než 10 let
```
Comment (dlouhak) [inline common.yml]: 10 let a méně

File: common.yml [hunk 2]
```diff
@@ -1003,3 +1127,11 @@ translations:
       t: >
         Lidé, které sledujete nebo jen chcete zmínit jejich práci (více jmen
         oddělujte čárkou).
+
+    # other surveys
+    - key: resources.other_surveys
+      t: Jiné průzkumy
+    - key: resources.other_surveys.description
+      t: Which other developer surveys do you participate in?
```
Comment (dlouhak) [inline common.yml]: Kterých dalších vývojářských průzkumů se účastníte?

File: common.yml [hunk 3]
```diff
@@ -77,16 +77,41 @@ translations:
       t: Nápověda k překladům
     - key: general.newly_added
       t: Nově přidané tento rok
+    - key: general.new
+      t: Nové
+    - key: general.devographics_banner
+      t: >
+        <a href="https://www.devographics.com/" target="_blank">Devographics</a>
+        je nový název kolektivu provozujícího průzkumy State of JavaScript,
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        je nový název kolektivu pořádajícího průzkumy State of JavaScript,
```

File: common.yml [hunk 4]
```diff
@@ -77,16 +77,41 @@ translations:
...
 
     # credits
     - key: credits.thanks
       t: Poděkování
+    - key: credits.contributors
+      t: Přispěvatelé
+    - key: credits.contributors.description
+      t: >
+        Zvláštní poděkování patří následujícím lidem za pomoc při
+        vytváření průzkumu.
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        Zvláštní poděkování za pomoc při
        vytváření průzkumu patří následujícím lidem.
```

File: common.yml [hunk 5]
```diff
@@ -188,6 +215,14 @@ translations:
       t: Stiskněte option/alt pro zapnutí režimu překladu.
     - key: general.charts_nivo
       t: Grafy vyrobeny s pomocí <a href="{link}">Nivo</a>.
+    - key: general.made_by_devographics
+      t: Vytvořili <a href="{link}">Devographics</a>
+    - key: general.view_code_example
+      t: Zobrazit ukázku kódu
+    - key: general.feature_code_example
+      t: "**{featureName}** Ukázka kódu"
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: "**{featureName}** ukázka kódu"
```

?

File: common.yml [hunk 6]
```diff
@@ -547,9 +533,26 @@ translations:
       t: Nebinární nebo nekonformní
     - key: options.gender.questioning
       t: V řešení
+    - key: options.gender.prefer_not_to_say
+      t: Raději nechci uvádět
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: Nechci uvádět
```

File: common.yml [hunk 7]
```diff
@@ -547,9 +533,26 @@ translations:
...
+    - key: options.gender.female.short
+      t: Žena
+    - key: options.gender.male.short
+      t: Muž
+    - key: options.gender.non_binary.short
+      t: Nebinární nebo nekonformní
+    - key: options.gender.questioning.short
+      t: V řešení
+    - key: options.gender.prefer_not_to_say.short
+      t: Raději nechci uvádět
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: Nechci uvádět
```

File: common.yml [hunk 8]
```diff
@@ -750,9 +794,31 @@ translations:
...
+    - key: options.bracket.round2
+      t: Vyhrává druhé kolo
+    - key: options.bracket.round3
+      t: Vyhrává třetí kolo
+    - key: options.bracket.combined
+      t: Kombinované
+
+    # email_temporary (receive updates?)
+    - key: options.email_temporary.yes
+      t: Ano, pošlete mi e-mail, až budou zveřejněny výsledky průzkumu
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: Ano, pošlete mi e-mail, jakmile budou výsledky průzkumu zveřejněny
```

File: common.yml [hunk 9]
```diff
@@ -826,6 +894,11 @@ translations:
       t: >
         Tato data používáme pro sledování a vylepšování vypovídací hodnoty
         průzkumu.
+    - key: user_info.gender.note
+      t: >
+        I když víme, že shromažďování a publikování dat o diverzitě může být
+        citlivou záležitostí, domníváme se, že je důležité získat tato data,
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        citlivou záležitostí, domníváme se, že je důležité získat tato data
```

File: common.yml [hunk 10]
```diff
@@ -835,6 +908,15 @@ translations:
       t: >
         Tato data používáme pro sledování a vylepšování vypovídací hodnoty
         průzkumu.
+    - key: user_info.race_ethnicity.note
+      t: >
+        I když víme, že shromažďování a publikování dat o diverzitě může být
+        citlivou záležitostí, domníváme se, že je důležité získat tato data,
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        citlivou záležitostí, domníváme se, že je důležité získat tato data
```

File: common.yml [hunk 11]
```diff
@@ -853,8 +935,21 @@ translations:
...
+    - key: user_info.email.note
+      t: >
+        Neuchováváme žádné osobní údaje, takže poté,
+        co vyplníte svou e-mailovou adresu, použijeme ji k vygenerování
+        jedinečného anonymizovaného identifikátoru a následně ji
+        vymažeme z našich záznamů.
     - key: user_info.country
       t: Země
+    - key: user_info.country.description
+      t: Kde v současné době bydlíte?
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: Kde v současné době žijete?
```

File: common.yml [hunk 12]
```diff
@@ -886,6 +981,26 @@ translations:
...
+      t: Jakou část průzkumu respondenti vyplnili?
+
+    # email_temporary (receive updates?)
+    - key: user_info.email_temporary
+      t: Chcete být informováni, až budou zveřejněny výsledky průzkumu?
+    - key: user_info.email_temporary.note
+      t: >
+        Neuchováváme žádná soukromá data, takže poté, co bude vaše e-mailová
+        adresa odeslána našemu poskytovateli mailing listu,
+        bude smazána z našich vlastních záznamů.
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        Neuchováváme žádná soukromá data. Jakmile bude vaše e-mailová
        adresa odeslána našemu poskytovateli mailing listu,
        bude smazána z našich záznamů.
```

File: common.yml [hunk 13]
```diff
@@ -853,8 +935,21 @@ translations:
...
+      t: >
+        Dotazník jste vyplnili anonymně. Pokud chcete,
+        můžete volitelně zadat svůj e-mail, abyste měli později přístup
+        ke svým datům a byli informováni o zveřejnění výsledků průzkumu.
+    - key: user_info.email.note
+      t: >
+        Neuchováváme žádné osobní údaje, takže poté,
+        co vyplníte svou e-mailovou adresu, použijeme ji k vygenerování
+        jedinečného anonymizovaného identifikátoru a následně ji
+        vymažeme z našich záznamů.
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
        Neuchováváme žádné osobní údaje. Jakmile
        vyplníte svou e-mailovou adresu, použijeme ji k vygenerování
        jedinečného anonymizovaného identifikátoru a následně ji
        vymažeme z našich záznamů.
```

File: common.yml [hunk 14]
```diff
@@ -1003,3 +1127,11 @@ translations:
...
         Lidé, které sledujete nebo jen chcete zmínit jejich práci (více jmen
         oddělujte čárkou).
+
+    # other surveys
+    - key: resources.other_surveys
+      t: Jiné průzkumy
+    - key: resources.other_surveys.description
+      t: Kterých dalších vývojářských průzkumů se účastníte?
+    - key: resources.other_surveys.others
+      t: Dodatečné výše neuvedené průzkumy
```
Comment (adamkudrna) [inline common.yml]:
```suggestion
      t: Jiné, výše neuvedené průzkumy
```

File: homepage.yml [hunk 15]
```diff
@@ -41,19 +23,56 @@ translations:
...
+    # newsletter
     - key: newsletter.stay_tuned
-      t: Zůstaňte v kontaktu
+      t: Nenechte si ujít další průzkum
     - key: newsletter.leave_your_email
       t: >
-        Zanechte nám svůj e-mail a my vám dáme vědět, jakmile se bude konat
-        další průzkum.
+        Zanechte nám svůj e-mail a my vám dáme vědět,
+        až proběhne další průzkum.
```
Comment (adamkudrna) [inline homepage.yml]:
```suggestion
        až bude probíhat další průzkum.
```

File: state_of_css.yml [hunk 16]
```diff
@@ -47,11 +47,29 @@ translations:
...
+    - key: sections.selectors.title
+      t: Selektory
+    - key: sections.selectors.description
+      t: CSS selektory.
+    - key: sections.colors.title
+      t: Barvy
+    - key: sections.colors.description
+      t: Vlastnosti související s barvami.
+    - key: sections.javascript_apis.title
+      t: JavaScript APIs
```
Comment (adamkudrna) [inline state_of_css.yml:69]:
```suggestion
      t: JavaScript API
```

or

```suggestion
      t: Javascriptová API
```

Also see below.

File: state_of_css.yml [hunk 17]
```diff
@@ -97,7 +122,21 @@ translations:
...
+      t: Ostatní CSS nástroje.
+
+    - key: sections.usage_css.title
+      t: Používání CSS
+    - key: sections.usage_css.description
+      t: Jak používáte CSS.
+    - key: sections.resources_css.title
+      t: Zdroje
+    - key: sections.resources_css.description
+      t: Jaké zdroje CSS konzultujete?
```
Comment (adamkudrna) [inline state_of_css.yml:139]: I'd prefer something like:

```suggestion
      t: V jakých zdrojích o CSS vyhledáváte?
```

Sounds more czech than _to consult sth._

File: state_of_css.yml [hunk 18]
```diff
@@ -145,130 +184,195 @@ translations:
...
+
+    - key: features.aria_attributes
+      t: ARIA HTML Attributes
+    - key: features.aria_attributes.description
+      t: <code>role</code>, <code>aria-label</code>, atd.
 
     # other features
     - key: features.variables
-      t: CSS proměnné (Custom Properties)
+      t: CSS Variables (Custom Properties)
```
Comment (adamkudrna) [inline state_of_css.yml:334]: Despite it's technically wrong to call it _variables_ (in any language), I prefer to translate this one.

File: state_of_css.yml [hunk 19]
```diff
@@ -521,7 +659,56 @@ translations:
...
+    - key: usage.interop_features.description
+      t: >
+        Kterou z těchto funkcí byste si nejvíce přáli, aby ji dodavatelé
+        prohlížečů vylepšili? (vyberte 3)
+    - key: usage.interop_features.note
+    # yamllint disable rule:line-length
+      t: >
+        [Interop 2023](https://github.com/web-platform-tests/interop/issues?q=is%3Aissue+is%3Aopen+label%3Afocus-area-proposal)
+        je koordinovaná snaha všech výrobců prohlížečů opravit
+        problémy s nekompatibilitou. Výsledky tohoto průzkumu pomohou
```
Comment (adamkudrna) [inline state_of_css.yml]:
```suggestion
        problémy s kompatibilitou. Výsledky tohoto průzkumu pomohou
```

File: state_of_css.yml [hunk 20]
```diff
@@ -531,8 +718,101 @@ translations:
...
+      t: >
+        Jak rozdělujete svůj čas mezi psaní CSS (včetně HTML markupu) a
+        JavaScript kódu?
+
+    ###########################################################################
+    # Resources
+    ###########################################################################
+
+    - key: resources.first_steps_css
+      t: First Steps With CSS
```
Comment (adamkudrna) [inline state_of_css.yml]:
```suggestion
      t: První kroky s CSS
```

File: surveys.yml [hunk 21]
```diff
@@ -8,6 +8,56 @@ translations:
...
+      t: Pokračovat jako host
+    - key: accounts.continue_as_guest.description
+      t: >
+        Pokud se průzkumu zúčastníte anonymně, po ukončení relace
+        **nebudete mít znovu přístup ke svým datům**.
+    - key: accounts.continue_as_guest.action
+      t: Pokračovat jako host
+
+    - key: accounts.magic_link.no_email
+      t: Prosím vyplňte váš e-mail
```
Comment (adamkudrna) [inline surveys.yml]:
```suggestion
      t: Prosím vyplňte svůj e-mail
```

File: surveys.yml [hunk 22]
```diff
@@ -85,33 +135,116 @@ translations:
...
+      # yamllint disable rule:line-length
       t: >
-        Co z následujícího vás nejlépe popisuje? Vyberte všechny relevantní
-        možnosti. Dočasná omezení nebo zranění mohou být také zahrnuta.
+        Další informace
+        [o loňském průzkumu naleznete zde](https://2021.stateofcss.com/en-US/about/).
+      # yamllint enable rule:line-length
 
+    - key: faq.survey_design_css2022
+      t: Jak byl navržen tento průzkum?
```
Comment (adamkudrna) [inline surveys.yml]:
```suggestion
      t: Jak byl tento průzkum navržen?
```

File: surveys.yml [hunk 23]
```diff
@@ -85,33 +135,116 @@ translations:
...
-        Tato data používáme ke sledování a vylepšování inkluzivity a vypovídací
-        hodnoty průzkumu. Tak jako všechny otázky, i tato je zcela dobrovolná.
+    - key: faq.data_used_css2022
+      t: Jak budou tato data použita?
+    - key: faq.data_used_css2022.description
+      t: >
+        Všechna shromážděná data budou zveřejněna. Stanou se tak zdrojem jak
+        pro vývojáře, kteří je konzultují, aby informovali o svých
+        technologických rozhodnutích, tak pro dodavatele prohlížečů,
+        kteří je používají k upřednostňování oblastí zájmu a vytváření plánů.
```
Comment (adamkudrna) [inline surveys.yml]:
```suggestion
        Všechna shromážděná data budou zveřejněna. Stanou se tak zdrojem jak
        pro vývojáře, kteří s jejich pomocí informují ostatní o svých
        technologických rozhodnutích, tak pro dodavatele prohlížečů,
        kteří je používají k prioritizaci oblastí zájmu a plánování vývoje.
```

**Review by adamkudrna — APPROVED** [review]

**Pull request merged**